lifelines/utils/ShiftedBetaGeometric.py

```diff
@@ -0,0 +1,726 @@
+            print("-"*35)
+
+        # Run likelihood optimization "restarts" times.
+        # noinspection PyTypeChecker
+        for step, guess in enumerate(initial_guesses):
+            # --- Variables
+            #   step: Integer - current step number
+            #  guess: Array - array with starting points for minimization
+
+            # --- Optimization
+            # Unbounded optimization (minimization) of negative log-likelihood
+            # Use a lambda function to make explicit the choice of variables.
+            # Notice that the variable p is an array of length 2 * n_params,
+            # which is split in half when passed to the _logp method. First
+            # half is alpha, second is beta.
+            new_opt = minimize(lambda p: self._logp(X=X,
+                                                    age=age,
+                                                    alive=alive,
+                                                    wa=p[:n_params],
+                                                    wb=p[n_params:]),
+                               guess,
+                               bounds=[(None, None)] * n_params * 2
+                               )
+
+            # For the first run only optimal is None, that being the case, we
+            # set the current values to both optimal (function value) as well
+            # as opt_params - parameters that minimize the function.
+            if optimal is None:
+                optimal = new_opt.fun
+                opt_params = new_opt.x
+
+            # Have we found a better value yet? If we have, update optimal
+            # with new function minimum and opt_params with corresponding
+            # minimizing parameters.
+            if new_opt.fun < optimal:
+                optimal = new_opt.fun
+                opt_params = new_opt.x
+
+            # Print current status if verbose is True.
+            if self.verbose:
+                print_string = "{step: {space}} | {elap_time:^10} | {func:13.5f} |"
+                print(print_string.format(step=step + 1,
+                                          elap_time=str(datetime.now() - start)[:-7],
+                                          func=float(optimal),
+                                          space=print_space))
+
+        # --- Parameter Values
+        # Optimization is complete, time to save best parameters.
+        # Note that we breakdown the parameters passed to and returned from
+        # the scipy.optimize.minimize object in two. The first half correspond
+        # to the alpha parameter, while the second half is beta.
+        self.alpha = opt_params[:n_params]
+        self.beta = opt_params[n_params:]
+
+        # --- Regularization Penalty
+        # Compute the regularization penalty applied to the parameter vectors.
+        # Remember that there is no penalty for bias!
+        reg_penalty = self.gammaa * sum(self.alpha[1:]**2) + \
+            self.gammab * sum(self.beta[1:]**2)
```

Comment:
this calculation is repeated in https://github.com/CamDavidsonPilon/lifelines/pull/232/files#diff-342dd249060066abe453c23a2bc99a23R212, maybe move it into it's own function?